Code/GraphMol/FileParsers/file_parsers_catch.cpp

```diff
@@ -6928,4 +6932,111 @@ TEST_CASE(
+
+  // get the largest fragment
+
+  unsigned int largestFragSize = 0;
+  ROMol *largestFrag = nullptr;
+  for (auto frag : frags) {
+    if (frag->getNumAtoms() > largestFragSize) {
+      largestFragSize = frag->getNumAtoms();
+      largestFrag = frag.get();
+    }
+  }
+
+  CHECK(largestFrag);
+  CHECK(getSubstanceGroups(*largestFrag).size() == fragTest.newSgroupCount);
+
+  if (fragTest.origSgroupCount == fragTest.newSgroupCount) {
+    // if the number of sgroups is the same, then the sgroups should be the
+    // same
+    for (unsigned int sgIndex = 0;
+         sgIndex < getSubstanceGroups(*largestFrag).size(); ++sgIndex) {
+      CHECK(getSubstanceGroups(*largestFrag)[sgIndex].getProp<std::string>(
+                "TYPE") ==
+            getSubstanceGroups(*mol)[sgIndex].getProp<std::string>("TYPE"));
+    }
+  }
+}
+
+TEST_CASE("FragmentSgroupTest", "[bug][reader]") {
+  std::string rdbase = getenv("RDBASE");
+  SECTION("basics") {
+    std::vector<FragTest> tests = {
+        FragTest("polymerSalt.mol", true, true, 1, 1),
+        FragTest("copolymer_sgroup.sdf", true, true, 1,
+                 0),  // fragmntation does not keep the sgroup for this one
+        FragTest("DataSgroup.sdf", true, true, 2, 2),
+        FragTest("DataSgroupMissingUnitsDisplayed.sdf", true, true, 1, 1),
+        FragTest("EmbeddedSGroupSUP_MUL.sdf", true, true, 2, 2),
+        FragTest("EmbeddedSgroupCOP_SUP.sdf", true, true, 2, 2),
+        FragTest("EmbeddedSgroupDAT_SUP.sdf", true, true, 2, 2),
+        FragTest("EmbeddedSgroupMUL_MUL.sdf", true, true, 3, 3),
+        FragTest("EmbeddedSgroupMUL_SUP.sdf", true, true, 2, 2),
+        FragTest("EmbeddedSgroupSRU_SUP.sdf", true, true, 2, 2),
+        FragTest("EmbeddedSgroupSUPEXP_SUP.sdf", true, true, 2, 2),
+        FragTest("EmbeddedSgroupSUPEXP_SUP2.sdf", true, true, 2, 2),
+        FragTest("EmbeddedSgroupSUP_SUP.sdf", true, true, 2, 2),
+        FragTest("EmbeddedSgroupSUP_SUP2.sdf", true, true, 2, 2),
+        FragTest("GenericSgroup.sdf", true, true, 1, 1),
+        FragTest("MarvinOldSuperGroupTest.sdf", true, true, 9, 5),
+        FragTest("MonomerSgroup.sdf", true, true, 1, 1),
+        FragTest("MultipleSgroup.sdf", true, true, 1, 1),
+        FragTest("MultipleSgroupParentInMiddleOfAtomBlock.sdf", true, true, 1,
+                 1),
+        FragTest("SgroupExpanded.sdf", true, true, 1, 1),
+        FragTest("SgroupMultAttach.sdf", true, true, 4, 4),
+        FragTest("Sgroup_MUL_ParentInMiddle.sdf", true, true, 1, 1),
+        FragTest("modification_sgroup.sdf", true, true, 2, 1),
+    };
+    for (auto test : tests) {
+      testFragmentation(test);
+    }
+  };
+}
+
+int main(int argc, char *argv[]) {
+  Catch::Session session;
+
+  if (argc > 2 && std::string(argv[2]) == "generate") {
+    generateExpectedFilesFlag = true;
+    --argc;
+  }
+
+  int returnCode = session.applyCommandLine(argc, argv);
+  if (returnCode != 0) return returnCode;
+  return session.run();
+}
```

Comment:
I guess this is no longer required, right?

Reply:
Correct - it is no longer required